sunpy/coordinates/transformations.py

```diff
@@ -260,7 +280,11 @@ def _transform_obstime(frame, obstime):
     # Transform to the new obstime using the appropriate loopback transformation
     new_frame = frame.replicate(obstime=obstime)
     if frame.obstime is not None:
-        return frame.transform_to(new_frame)
+        if _apply_diffrot:
+            from .metaframes import RotatedSunFrame  # avoid a circular import
+            return RotatedSunFrame(base=frame, rotated_time=obstime).transform_to(new_frame)
+        else:
+            return frame.transform_to(new_frame)
```

Comment:
You don't really need the else, here. :shrug: 
```suggestion
        return frame.transform_to(new_frame)
```